v2/tools/generator/internal/astmodel/package_definition.go

```diff
@@ -175,6 +178,19 @@ func (p *PackageDefinition) writeTestFile(
 	return nil
 }
 
+func (p *PackageDefinition) createVersion(ref InternalPackageReference) string {
```

Comment:
minor: No actual usage of `p` receiver in this function, you could instead make this a pure function and change `NewPackageDefinition` above to do the following:
```
	return &PackageDefinition{
		PackageName: ref.PackageName(),
		GroupName:   ref.Group(),
		Version: createVersion(ref),
		Path:        ref.FolderPath(),
		definitions: make(TypeDefinitionSet),
	}
```

or, you could make this a method on `InternalPackageReference`, though I know it's an interface so I guess you can't easily put a method on it (unless you promoted Version() to be on the interface). You could make it a public `astmodel` method instead if you prefer: `PackageVersion(ref InternalPackageReference)` or similar as well.

It feels like it doesn't really make sense on `PackageDefinition` though., and having it on PackageDefinition means you end up having to do the awkward "construct a partial package definition, then fill the rest out" in the `New` function, which is _fine_ (it is `New` after all so the contract is just that it's fully specified when it returns) but it still feels more awkward than it needs to be.

Reply:
I prefer it with the receiver, because that acts as a namespace to restrict access. Moving it to a pure function is inviting use from other locations, which is inappropriate.